### PR TITLE
feat(ses): `grant` methods to `IEmailIdentity`

### LINKED ISSUE #29083: aws-ses: add `grant` methods to `EmailIdentity`
### Describe the feature

Add a general `grant` method to `EmailIdentity` objects, as well as a more specific method for the send email action.

### Use Case

When granting send email access to a lambda the grant needs to be constructed manually, including constructing the ARN for the identity. e.g.
```
  Grant.addToPrincipal({
    grantee,
    actions: ["ses:SendEmail"],
    resourceArns: [
      this.stack.formatArn({
        service: 'ses',
        resource: 'identity',
        resourceName: 'test@example.com',
      }),
    ],
    scope: this
  })
```
This is dissimilar to other constructs, which generally expose a `grant` method and one or more convenience methods for particularly relevant groups of actions.

### Proposed Solution

Implement a `grant` and a `grantSendEmail` method on `EmailIdentity` (and `IEmailIdentity`), so that the above use can be accomplished with a simple
```
    const emailIdentity = new EmailIdentity(this, 'Identity', {
      identity: Identity.email('test@example.com')
    });
    emailIdentity.grantSendEmail(grantee);
```

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [ ] This feature might incur a breaking change

### CDK version used

2.126.0

### Environment details (OS name and version, etc.)

Ubuntu 22.04

### PR DESCRIPTION
**Pull request by karaken12**
### Issue

Closes #29083

### Reason for this change

When granting send email access to a lambda the grant needs to be constructed manually, including constructing the ARN for the identity. e.g.

```
  Grant.addToPrincipal({
    grantee,
    actions: ["ses:SendEmail"],
    resourceArns: [
      this.stack.formatArn({
        service: 'ses',
        resource: 'identity',
        resourceName: 'test@example.com',
      }),
    ],
    scope: this
  })
```

This is dissimilar to other constructs, which generally expose a grant method and one or more convenience methods for particularly relevant groups of actions.

### Description of changes

Added `grant` and `grantSendEmail` to `IEmailIdentity`, and added a common abstract class, `BaseEmailIdentity` with the relevant grant code. This is to avoid code duplication between the full `EmailIdentity` and the `Import` class.

### Description of how you validated changes

Tests added for grants on both new and imported email identities, and a test to validate the `grantSendEmail` method.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by kaizencc — APPROVED** [review]
Thanks @karaken12, this looks good!

**Comment by karaken12** [comment]
> Thanks @karaken12, this looks good!

Great, thanks! I can still add an integration test if you want?

**Comment by karaken12** [comment]
Updated an integration test (as the bot requested!), corrected a documentation error, and updated the list of actions in the `grantSendEmail` convenience method.

File: packages/aws-cdk-lib/aws-ses/README.md [hunk 1]
```diff
@@ -210,6 +210,21 @@ for (const record of identity.dkimRecords) {
 }
 ```
 
+#### Grants
+
+To grant a specific action to a principal use the `grant` method.
+For sending emails, `grantSendEmail` can be used instead:
+
+```ts
+declare const user: iam.User;
```
Comment (kaizencc) [inline packages/aws-cdk-lib/aws-ses/README.md:219]:
```suggestion
```ts
import * as iam from 'aws-cdk-lib/aws-iam';
declare const user: iam.User;
```

**Review by kaizencc — APPROVED** [review]

**Pull request merged**